source/test/TestApp/PlayFabGroupsTest.cpp

```diff
@@ -0,0 +1,87 @@
+
+#include "TestAppPch.h"
+
+#include "PlayFabGroupsTest.h"
+#include "playfab/PlayFabApiSettings.h"
+#include "playfab/PlayFabSettings.h"
+
+using namespace PlayFab;
+
+namespace PlayFabUnit
+{
+
+    // GROUPS API
+    void PlayFabGroupsTest::GroupsApiTest(TestContext& testContext)
+    {
+        auto req = ClientModels::LoginWithCustomIDRequest();
+        req.CustomId = PlayFabSettings::buildIdentifier;
+
+        clientApi->LoginWithCustomID(req, Callback(&PlayFabGroupsTest::GroupsTestLoginCallback), Callback(&PlayFabGroupsTest::GroupsTestSharedFailureCallback), &testContext);
+    }
+
+    void PlayFabGroupsTest::GroupsTestLoginCallback(const ClientModels::LoginResult& result, void* customData)
+    {
+        auto groupApi = std::make_shared<PlayFabGroupsInstanceAPI>(result.authenticationContext);
```

Comment:
Move this into the ClassSetup, move the groupApi variable into the test class, and construct them both with shared settings and context.
I think that's why you were having such issues with the titleID and whatnot.
This one is being created with staticSettings, and clientApi is being created with a custom context, so they're not linked to the same information.

You're constructing the API instances completely differently, and that's the source of your data desync.
Your solution of result.authenticationContext HELPS, but it fails to share settings.
It's better to just create them together in ClassSetup.